eng/native/configuretools.cmake

```diff
@@ -91,3 +92,17 @@ if (NOT CLR_CMAKE_HOST_WIN32)
     set(LD_OSX 1)
   endif()
 endif()
+
+if (TARGET CLR_CMAKE_HOST_LINUX)
+  execute_process(
+    COMMAND bash -c "if ${CMAKE_STRINGS} \"${CMAKE_SYSROOT}/usr/bin/ldd\" 2>&1 | grep -q musl; then echo musl; fi"
+    OUTPUT_VARIABLE CLR_CMAKE_LINUX_MUSL
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if(CLR_CMAKE_LINUX_MUSL STREQUAL musl)
+    set(CLR_CMAKE_HOST_LINUX_MUSL 1)
+    set(CLR_CMAKE_TARGET_UNIX 1)
+    set(CLR_CMAKE_TARGET_LINUX 1)
+    set(CLR_CMAKE_TARGET_LINUX_MUSL 1)
+  endif()
+endif()
```

Comment:
would you mind moving this back to configureplatforms.cmake? configuretools.cmake is about defining the tools, not the platforms.

Reply:
Added a comment explaining why it is deferred to configuretools.

> This introspection depends on CMAKE_STRINGS, whcih is why it's in this file instead of configureplatform

Reply:
ok, I'm just a bit concerned about setting the platform variables here (nobody will know to also look into configuretools.cmake when modifying configureplatform.cmake).

do we need the CLR_CMAKE_TARGET_UNIX and CLR_CMAKE_TARGET_LINUX here? shouldn't it already be set by configureplatform.cmake?

Reply:
We currently do not include `configureplatforms.cmake` separately outside the `eng/native` dir. The inter-dependencies of these files sometimes get tricky, like in this case. tools need platforms introspection and vice-versa. Had we not removed binutils from prereq images, we would have kept this condition in `configureplatforms.cmake`. So this is potentially a workaround which we can attribute to CBL-mariner change. (BTW, I am happy that we are fully utilizing the same toolchain; llvm or gcc/binutils without mixing them).

Perhaps we can merge these two files if file naming is becoming confusing as we are no longer cleanly splitting the code on tools/platforms boundary.

Reply:
> Perhaps we can merge these two files if file naming is becoming confusing as we are no longer cleanly splitting the code on tools/platforms boundary.

Yeah that sounds like a good idea, it can be done in a separate PR :)